plugins/source/aws/codegen/recipes/ram.go

```diff
@@ -0,0 +1,168 @@
+					Type:     schema.TypeString,
+					Resolver: `schema.PathResolver("AssociatedEntity")`,
+					Options:  pk,
+				},
+				codegen.ColumnDefinition{
+					Name:        "resource_share_arn",
+					Type:        schema.TypeString,
+					Resolver:    `schema.PathResolver("ResourceShareArn")`,
+					Description: "Resource Share ARN",
+					Options:     pk,
+				},
+			),
+		},
+		{
+			SubService: "resource_share_associated_resources",
+			Struct:     new(types.ResourceShareAssociation),
+			Multiplex:  "", // it's a relation for resource_shares
+			SkipFields: []string{"AssociatedEntity", "ResourceShareArn"},
+			ExtraColumns: append(defaultRegionalColumns,
+				codegen.ColumnDefinition{
+					Name:     "associated_entity",
+					Type:     schema.TypeString,
+					Resolver: `schema.PathResolver("AssociatedEntity")`,
+					Options:  pk,
+				},
+				codegen.ColumnDefinition{
+					Name:        "resource_share_arn",
+					Type:        schema.TypeString,
+					Resolver:    `schema.PathResolver("ResourceShareArn")`,
+					Description: "Resource Share ARN",
+					Options:     pk,
+				},
+			),
+		},
+		{
+			SubService: "resource_share_invitations",
+			Struct:     new(types.ResourceShareInvitation),
+			Multiplex:  mx,
+			SkipFields: []string{"ResourceShareInvitationArn"},
+			ExtraColumns: append(defaultRegionalColumns,
+				codegen.ColumnDefinition{
+					Name:     "arn",
+					Type:     schema.TypeString,
+					Resolver: `schema.PathResolver("ResourceShareInvitationArn")`,
+					Options:  pk,
+				},
+			),
+			ResolverAndMockTestTemplate: "get_resources_paginated_1",
+		},
+		{
+			SubService: "resource_share_permissions",
+			Struct:     new(types.ResourceSharePermissionSummary),
+			Multiplex:  mx,
+			SkipFields: []string{"Arn"},
+			ExtraColumns: append(defaultRegionalColumns,
+				codegen.ColumnDefinition{
+					Name:     "arn",
+					Type:     schema.TypeString,
+					Resolver: `schema.PathResolver("Arn")`,
+					Options:  pk,
+				},
+				codegen.ColumnDefinition{
+					// grabbed from types.ResourceSharePermissionDetail
+					Name:     "permission",
+					Type:     schema.TypeJSON,
+					Resolver: `resolveResourceSharePermissionDetailPermission`,
+				},
+			),
+		},
+		{
+			SubService: "resource_types",
+			Struct:     new(types.ServiceNameAndResourceType),
+			Multiplex:  mx,
+			SkipFields: []string{"ResourceType", "ServiceName"},
+			ExtraColumns: append(defaultRegionalColumns,
+				codegen.ColumnDefinition{
+					Name:     "resource_type",
+					Type:     schema.TypeString,
+					Resolver: `schema.PathResolver("ResourceType")`,
+					Options:  pk,
+				},
+				codegen.ColumnDefinition{
+					Name:     "service_name",
+					Type:     schema.TypeString,
+					Resolver: `schema.PathResolver("ServiceName")`,
+					Options:  pk,
+				},
+			),
+		},
+	}
+	for _, r := range resources {
+		r.Service = "ram"
+		r.Description = "https://docs.aws.amazon.com/ram/latest/APIReference/API_" + r.StructName() + ".html"
```

Comment:
Why not get rid of `mx` above and put it as a single line here?

Reply:
`mx` is not used for the relations (naturally, you don't need to do multiplexing as the parent will reside in the same region).